### PR TITLE
Email after volunteering or unenrolling

### LINKED ISSUE #372: Email after volunteering or unenrolling
The app should send the user a confirmation email after they volunteer for and/or unenroll from an event.

Labels: "question" (do we want to send people lots of emails, @OhMcGoo?), "enhancement."

### PR DESCRIPTION
**Pull request by mmoore99**
Fixes #372

Confirmation emails will now be sent to the volunteer upon signing up for or unenrolling from an activity.

Note, as discussed in the issue comments, this PR does not provide for allowing user to opt out of receiving these emails.

### CODE REVIEW

**Comment by tonysurma** [comment]
@MisterJames can someone take a look at this PR before I merge?

File: AllReadyApp/Web-App/AllReady/Features/Notifications/NotifyVolunteerForActivitySignup.cs [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+            _options = options;
+        }
+
+        public void Handle(VolunteerInformationAdded notification)
+        {
+            var model = _bus.Send(new ActivityDetailForNotificationQuery {ActivityId = notification.ActivityId});
+
+            var signup = model.UsersSignedUp?.FirstOrDefault(s => s.User.Id == notification.UserId);
+
+            if (string.IsNullOrWhiteSpace(signup?.PreferredEmail)) return;
```
Comment (mheggeseth) [inline AllReadyApp/Web-App/AllReady/Features/Notifications/NotifyVolunteerForActivitySignup.cs]: Fall back to `User.Email` if no `PreferredEmail` (potentially also require `User.EmailConfirmed` but I don't think that's strictly necessary yet)

File: AllReadyApp/Web-App/AllReady/Features/Notifications/NotifyVolunteerForUserUnenrolls.cs [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+            _options = options;
+        }
+
+        public void Handle(UserUnenrolls notification)
+        {
+            var model = _bus.Send(new ActivityDetailForNotificationQuery {ActivityId = notification.ActivityId});
+
+            var signup = model.UsersSignedUp?.FirstOrDefault(s => s.User.Id == notification.UserId);
+
+            if (string.IsNullOrWhiteSpace(signup?.PreferredEmail)) return;
```
Comment (mheggeseth) [inline AllReadyApp/Web-App/AllReady/Features/Notifications/NotifyVolunteerForUserUnenrolls.cs]: Fall back to `User.Email` if no `PreferredEmail` (potentially also require `User.EmailConfirmed` but I don't think that's strictly necessary yet)

**Comment by mmoore99** [comment]
Submitted changes as requested in PR review by @mheggeseth

**Comment by mheggeseth** [comment]
:shipit:

**Pull request merged**